spec/factorial_spec.rb

```diff
@@ -0,0 +1,16 @@
+require 'rspec'
+require_relative '../solver'
+
+describe Solver do
+  describe '#solver?' do
+    it 'return factorial of n' do
+      solver = Solver.new
+      expect(solver.factorial(3)).to eql(6)
+    end
+
+    it 'return factorial of n' do
+      solver = Solver.new
+      expect(solver.factorial(5)).to eql(120)
+    end
+  end
+end
```

Comment:
- [ ] Please add a negative case where the test should raise an exception ```raise_exception```
![image](https://github.com/Kaghenimbale/test_driven_development/assets/62705947/42c55896-028b-4fa7-815c-ec886fd0bfd9)

Reply:
- [ ] Please add a test case when the N is 0 which is a special case and it should return 1